### PR TITLE
Fix for validity domain checker

### LINKED ISSUE #509: ValidityDomainChecker checks even when the associated components is not used
**Describe the bug**
If two components use the same value name and have different value for allowable range in the ValidityDomainChecker but are not used at the same time, the ValidityDomainChecker of the components not used in the problem will still trigger  and raise a warning

**To Reproduce**
This code should reproduce the bug: 
[bug_validity_domain_checker.zip](https://github.com/fast-aircraft-design/FAST-OAD/files/14525853/bug_validity_domain_checker.zip)

**Expected behavior**
Only the ValidityDomainChecker of components actually used should proc

**Environment**
- Operating System: Windows 10
- Python environment: Anaconda Python 3.8.13
- Packages: 
aenum==3.1.12
alabaster==0.7.13
anyio==3.6.2
argcomplete==3.0.5
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
astroid==2.11.7
atomicwrites==1.4.1
attrs==23.1.0
Babel==2.12.1
backcall==0.2.0
beautifulsoup4==4.12.2
black==21.9b0
bleach==6.0.0
brotlipy==0.7.0
build==0.10.0
CacheControl==0.12.6
cachy==0.3.0
certifi==2023.5.7
cffi==1.15.1
cfgv==3.3.1
charset-normalizer==3.1.0
cleo==0.8.1
click==8.0.4
clikit==0.6.2
colorama==0.4.6
coverage==7.2.6
crashtest==0.3.1
cryptography==37.0.1
cycler==0.11.0
data==0.4
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.13
dill==0.3.6
distlib==0.3.6
docutils==0.17.1
ensure==1.0.3
entrypoints==0.4
et-xmlfile==1.1.0
fast-oad-core==1.6.0
fast-oad-cs23==1.2.0
fastjsonschema==2.17.1
filelock==3.12.0
flake8==4.0.1
fonttools==4.38.0
funcsigs==1.0.2
future==0.18.2
html5lib==1.1
identify==2.5.24
idna==3.4
imagesize==1.4.1
importlib-metadata==4.2.0
importlib-resources==5.12.0
iniconfig==2.0.0
iPOPO==1.0.1
ipykernel==6.16.2
ipysheet==0.7.0
ipython==7.34.0
ipython-genutils==0.2.0
ipywidgets==7.7.5
isort==5.11.5
jedi==0.18.2
Jinja2==3.1.2
joblib==1.2.0
json5==0.9.14
jsonrpclib-pelix==0.4.3.2
jsonschema==4.17.3
jupyter_client==7.4.9
jupyter_core==4.12.0
jupyter-server==1.24.0
jupyterlab==3.5.3
jupyterlab-pygments==0.2.2
jupyterlab_server==2.15.2
jupyterlab-widgets==1.1.4
keyring==23.4.0
kiwisolver==1.4.4
latex==0.7.0
latexcodec==2.0.1
lazy-object-proxy==1.9.0
llvmlite==0.41.1
lockfile==0.12.2
lxml==4.9.2
MarkupSafe==2.1.2
matplotlib==3.5.3
matplotlib-inline==0.1.6
mccabe==0.6.1
mistune==2.0.5
mpmath==1.2.1
msgpack==1.0.3
mypy-extensions==1.0.0
nbclassic==1.0.0
nbclient==0.7.4
nbconvert==7.4.0
nbformat==5.8.0
nbval==0.9.6
nest-asyncio==1.5.6
networkx==2.6.3
nodeenv==1.8.0
notebook==6.5.4
notebook_shim==0.2.3
numba==0.58.1
numpy==1.24.4
openmdao==3.26.0
openmdao-extensions==1.2.3
openpyxl==3.0.10
packaging==23.1
pandas==1.5.3
pandocfilters==1.5.0
parso==0.8.3
pastel==0.2.1
pathspec==0.11.1
pickleshare==0.7.5
Pillow==9.5.0
Pint==0.19.2
pip==22.2.2
pipx==1.2.0
pkginfo==1.9.6
pkgutil_resolve_name==1.3.10
platformdirs==2.6.2
plotly==5.14.1
pluggy==1.0.0
poetry==1.1.7
poetry-core==1.0.3
pre-commit==2.21.0
prometheus-client==0.16.0
prompt-toolkit==3.0.38
psutil==5.9.5
py==1.11.0
pybtex==0.24.0
pybtex-docutils==1.0.2
pycodestyle==2.8.0
pycparser==2.21
pyDOE2==1.3.0
pyflakes==2.4.0
Pygments==2.15.1
pylev==1.3.0
pylint==2.13.9
pyOpenSSL==22.0.0
pyparsing==3.1.1
pyproject_hooks==1.0.0
pyrsistent==0.19.3
PySocks==1.7.1
pytest==6.2.5
pytest-cov==3.0.0
python-dateutil==2.8.2
pytz==2023.3
pyvplm==0.1.0
pywin32==306
pywin32-ctypes==0.2.0
pywinpty==2.0.10
PyYAML==6.0
pyzmq==25.0.2
regex==2023.5.5
requests==2.31.0
requests-toolbelt==0.9.1
ruamel.yaml==0.17.26
ruamel.yaml.clib==0.2.7
scikit-learn==1.1.2
scipy==1.10.1
Send2Trash==1.8.2
setuptools==67.8.0
setuptools-scm==6.4.2
shellingham==1.3.1
shutilwhich==1.1.0
six==1.16.0
sniffio==1.3.0
snowballstemmer==2.2.0
soupsieve==2.4.1
Sphinx==4.3.2
sphinx-rtd-theme==1.2.1
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-bibtex==2.5.0
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jquery==4.1
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
stdatm==0.2.0
sympy==1.11.1
tabulate==0.8.10
tempdir==0.7.1
tenacity==8.2.2
terminado==0.17.1
threadpoolctl==3.1.0
tinycss2==1.2.1
tokenize-rt==5.0.0
toml==0.10.2
tomli==1.2.3
tomli_w==1.0.0
tomlkit==0.11.8
tornado==6.2
traitlets==5.9.0
typing_extensions==4.6.1
urllib3==2.0.2
userpath==1.8.0
virtualenv==20.16.2
wcwidth==0.2.6
webencodings==0.5.1
websocket-client==1.5.2
wheel==0.40.0
widgetsnbextension==3.6.4
win-inet-pton==1.1.0
wincertstore==0.2
wop==2.5.0
wrapt==1.15.0
xdsmjs==2.0.0
xlrd==2.0.1
xlwt==1.3.0
zipp==3.15.0

### PR DESCRIPTION
**Pull request by christophe-david**
This PR resolves #509.

As soon as a component was decorated with `ValidityDomainChecker`, it was included in the final check, regardless of the presence of the component in the run problem.

A flag is now added to activate or not the checks by component.

### CODE REVIEW

File: src/fastoad/openmdao/validity_checker.py
Comment (florentLutz) [inline src/fastoad/openmdao/validity_checker.py:1]: In the comments describing the class, there are code blocks to illustrate how the decorator should be used. In those code block the `ExplicitComponent` class for OpenMDAO is not properly written

File: src/fastoad/openmdao/validity_checker.py [hunk 1]
```diff
@@ -160,6 +160,8 @@ def check_problem_variables(cls, problem: om.Problem) -> List[CheckRecord]:
         :param problem:
         :return: the list of checks
```
Comment (florentLutz) [inline src/fastoad/openmdao/validity_checker.py:161]: I have a question on the `__call__` method, when is it used ?

File: src/fastoad/openmdao/validity_checker.py [hunk 2]
```diff
@@ -169,18 +171,23 @@ def check_problem_variables(cls, problem: om.Problem) -> List[CheckRecord]:
...
     @classmethod
-    def check_variables(cls, variables: VariableList) -> List[CheckRecord]:
+    def check_variables(
+        cls, variables: VariableList, activated_only: bool = True
+    ) -> List[CheckRecord]:
         """
         Check values of provided variables against registered limits.
 
         :param variables:
+        :param ignore_deactivated:
```
Comment (florentLutz) [inline src/fastoad/openmdao/validity_checker.py:181]: The name of the option in the comment does not match the signature of the method. Additionally, wht does this parameter does ?

File: src/fastoad/openmdao/tests/test_validity_checker.py [hunk 3]
```diff
@@ -230,3 +233,27 @@ def setup(self):
     ValidityDomainChecker.log_records(records)
     with open(log_file_path) as log_file:
         assert len(log_file.readlines()) == 4
+
+
+def test_sellar(caplog):
```
Comment (florentLutz) [inline src/fastoad/openmdao/tests/test_validity_checker.py:238]: How does this caplog works ? I can't see it defined anywhere or imported.

Comment (florentLutz) [inline src/fastoad/openmdao/tests/test_validity_checker.py:238]: Another comment, which is rther a question. When I click on the little triangle to run only this tests in PyCharm, 3046 test are collected and ran. Although when I run the file hat contains this test, only 3 are collected. Do you have any idea why that might happen ?

**Review by florentLutz — COMMENTED** [review]
Thank you for the quick fix ! Only a few questions and comments but only to figure out how things work :smile:

File: src/fastoad/openmdao/validity_checker.py (hunk 2 above)
Comment (christophe-david) [reply src/fastoad/openmdao/validity_checker.py]: doscstring fixed in 1fbf6a528fbb9bbb0b216c36139cfa0023be121f

**Review by florentLutz — APPROVED** [review]

File: src/fastoad/openmdao/validity_checker.py
Comment (christophe-david) [reply src/fastoad/openmdao/validity_checker.py:1]: fixed in 4e364c54a4a22c420ab646cb70fa78ba72d817da

**Review by florentLutz — APPROVED** [review]

**Pull request merged**